presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -349,6 +356,37 @@ protected Node visitFunctionCall(FunctionCall node, Void context)
                 rewrittenArguments.build());
     }
 
+    /**
+     * This is special-cased for now as COUNT is the only non-associative
+     * function supported by MV rewrites. In the future, we will want to
+     * support more non-associative functions and explore more
+     * extensible options.
+     *
+     * Functions in optimized MV queries are by default expanded to
+     * func(column_derived_from_func_in_mv). This only works for associative
+     * functions. Count is non-associative: COUNT(x U y) != COUNT(Count(x), COUNT(y)).
```

Comment:
it's better to use ascii code: `\union`

Reply:
I'm sure I'm doing something wrong, but this causes compilation failure `illegal unicode escape`: https://github.com/prestodb/presto/runs/6834514618?check_suite_focus=true . I'm putting a space in front of the backslash as described [here](https://stackoverflow.com/a/3949144), but I want to check that's what you had in mind.